.dockerignore

```diff
@@ -1,6 +1,10 @@
 .git
+.github
+test
 .dockerignore
-circle.yml
+.gitignore
+*.yml
+Dockerfile*
+LICENSE
```

Comment:
Legally speaking, isn't it a license violation to not put the LICENSE file inside the Docker image?

The license says:
> The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.


IANAL but I would say the Docker Image qualifies as a copy of the software and should thus contain a copy of the license.

Reply:
Yes, I think it should be included (I thought I left a comment about that, but must've been another repository 😓)

Reply:
Should it be included at the root of the image or inside `/app` ?

Reply:
To be honest; it's something that's often forgotten, and not sure if there's a clearly defined location for it, although for regular `deb` / `rpm` packages, such files (`LICENSE`, `NOTICE`) would be installed in (e.g.) `/usr/share/doc/<PACKAGE NAME>/`.


In addition, it's also possible to set one or more `LABEL` to provide some metadata (see https://github.com/opencontainers/image-spec/blob/main/annotations.md#pre-defined-annotation-keys), although this can be somewhat tricky if the image is based on a base image and contains other software (with different licensing), so not sure if adding "partial" information is better than "not including it".

Reply:
The GitHub Actions workflow uses `docker/metadata-action@v3` to tag the image, so we're already setting some `org.opencontainers.image` labels on the images, including `licenses` :

```
org.opencontainers.image.title=nginx-proxy
org.opencontainers.image.description=Automated nginx proxy for Docker containers using docker-gen
org.opencontainers.image.url=https://github.com/nginx-proxy/nginx-proxy
org.opencontainers.image.source=https://github.com/nginx-proxy/nginx-proxy
org.opencontainers.image.version=latest
org.opencontainers.image.created=2022-03-07T15:23:02.325Z
org.opencontainers.image.revision=6b48e11e5cbe8066a619300fa7b40408da02387c
org.opencontainers.image.licenses=MIT
```

I really have no preference over putting the `LICENSE` file at the root or inside `/app`.

Reply:
The GitHub Actions workflow uses `docker/metadata-action@v3` to tag the image, so we're already setting some `org.opencontainers.image` labels on the images, including `licenses` :

```
org.opencontainers.image.title=nginx-proxy
org.opencontainers.image.description=Automated nginx proxy for Docker containers using docker-gen
org.opencontainers.image.url=https://github.com/nginx-proxy/nginx-proxy
org.opencontainers.image.source=https://github.com/nginx-proxy/nginx-proxy
org.opencontainers.image.version=latest
org.opencontainers.image.created=2022-03-07T15:23:02.325Z
org.opencontainers.image.revision=6b48e11e5cbe8066a619300fa7b40408da02387c
org.opencontainers.image.licenses=MIT
```

I really have no preference over putting the `LICENSE` file at the root or inside `/app`.

Reply:
Ah! Didn't check if it was using the metadata-action, nice!

Yes, no preference either where to stash the file(s) inside the image.

Reply:
> not arguing against it but not including the License inside the Docker image does not seem to bother other MIT licensed projects like Traefik or Dokku.

I can't speak for other projects, but according to the license the full text and copyright notice should be included in the image. 
From what I can see the metadata-action only puts in the label 'MIT' and links to the repository - this isn't the same.
My personal preference is to include the LICENSE file directly inside the image.

Reply:
@tkw1536 @thaJeztah done in 4ea3437dfab423b6710b32b2aa9d50f9f750bed9